local-storage/src/provider.rs

```diff
@@ -75,13 +77,64 @@ impl SqliteStorageProvider {
 
         Ok(())
     }
+
+    fn get_blocks_recursive_query(
+        &self,
+        cid: &str,
+        offset: Option<u32>,
+        window_size: Option<u32>,
+    ) -> Result<Vec<StoredBlock>> {
+        let mut base_query = "
+        WITH RECURSIVE cids(x,y,z) AS (
```

Comment:
I was not familiar with `WITH RECURSIVE`. That's interesting.
How is performance? Is this function (regardless of how it's implemented) a memory limitation concern?

Reply:
I haven't run into any show stopping performance issues with files in the 5-50 mB range, but also haven't done much memory profiling around larger files here. I think we are Ok for the smaller files scheduled for usage in the demonstration.